eng/versioning/scan_for_unreleased_dependencies.ps1

```diff
@@ -87,6 +87,21 @@ Get-ChildItem -Path $serviceDirectory -Filter pom*.xml -Recurse -File | ForEach-
                         Write-Error-With-Color "Error: Cannot release non-beta libraries with beta_ dependencies. dependency=$($versionUpdateTag)"
                     }
                     continue
+                } else {
+                    # If this is an external dependency then continue
+                    if ($versionUpdateTag -match "external_dependency}") {
```

Comment:
I think it is actually a good idea to do a check for both internal and external dependencies. I'd suggest we use the SemVer.ps1 script we have under common and check to see if our version and if it is not prerelease then we should error for any dependency that is marked as a prerelease.

Reply:
@weshaggard, sure. I can look and see if this is feasible for external_dependencies on Monday. There are, unfortunately, beta external dependencies. Also, I'd have to verify that the SemVer.ps1 does correctly parse all of the external dependency versions first as Maven is a bit more liberal with what it allows for versions.

Reply:
I hope we don't depend on external prereleases in our GA packages that will definitely be a recipe for potential disaster.

Reply:
We do but only in a <test> scope. I'm less concerned about external dependencies since the maven-enforcer-plugin was added and every dependency is now being scrutinized (track 2 only though).